frontend/Dockerfile

```diff
@@ -1,35 +1,52 @@
 # --- Copy source files to Docker ---
 FROM node:18-alpine AS base
-WORKDIR /usr/src/app
+WORKDIR /opt
 
 # Install packages and cache the results
 # so we don't have to install everything when source code changes
-COPY frontend/package.json .
-COPY frontend/yarn.lock .
-RUN yarn
+COPY package.json .
+COPY yarn.lock .
+RUN yarn install --frozen-lockfile
+
+FROM base AS shared
+
+WORKDIR /opt
+
+COPY shared/ ./shared
+RUN yarn workspace vaa-shared install --frozen-lockfile
+RUN yarn workspace vaa-shared build
+
+FROM base AS frontend
+
+WORKDIR /opt
 
 # Copy files from project root to Docker image
-COPY frontend .
-COPY shared /usr/src/app/src/shared
+COPY --from=shared /opt/shared ./shared
+COPY frontend/ ./frontend
+RUN yarn workspace vaa-frontend install --frozen-lockfile
+
+FROM frontend AS development
+
+WORKDIR /opt
 
-FROM base AS development
 EXPOSE 5173
+CMD yarn workspace vaa-frontend dev --host
 
-CMD yarn dev --host
+FROM frontend AS build
 
-FROM base AS build
-RUN yarn build
+RUN yarn workspace vaa-frontend build
 
 FROM node:18-alpine AS production
 ARG NODE_ENV=production
 ENV NODE_ENV=${NODE_ENV}
-WORKDIR /usr/src/app
 
-COPY --from=build ./usr/src/app/package.json .
-COPY --from=build ./usr/src/app/yarn.lock .
-COPY --from=build ./usr/src/app/build .
+WORKDIR /opt
+
+COPY --from=build ./opt/frontend/package.json ./frontend
+COPY --from=build ./opt/yarn.lock .
+COPY --from=build ./opt/frontend/build .
 
-RUN yarn install --production
+RUN yarn workspace vaa-frontend install --production --frozen-lockfile
```

Comment:
Strangely enough, this command fails when deploying to Render, but not others involving workspaces.

Logs filtered by "workspace":

<img width="1728" alt="Screenshot 2024-08-09 at 19 06 48" src="https://github.com/user-attachments/assets/1e5e5a46-58a7-4732-a820-2e5ca4998bad">

The whole command 23:

<img width="1728" alt="Screenshot 2024-08-09 at 19 07 07" src="https://github.com/user-attachments/assets/e4fa1f35-4fde-4095-acf4-21340c212a10">

Reply:
Great, this works now!